executor/include/execution/trait/is_executor.hpp

```diff
@@ -5,28 +5,31 @@
 #pragma once
 
 #include <execution/executor/base_executor.hpp>
+#include <execution/trait/common_traits.hpp>
 #include <experimental/type_traits>
 #include <type_traits>
 
 namespace execution {
 
-template <class Executor, typename Function = void (*)(), typename = void>
+template <class Executor, typename = void>
 struct is_executor : std::false_type {};
 
 namespace detail {
 const auto noop = [] {};
 }
 
-template <typename Executor, typename Function>
+template <typename Executor>
 struct is_executor<
-    Executor, Function,
+    Executor,
     std::enable_if_t<
-        std::is_copy_constructible_v<
-            std::remove_cv_t<std::remove_reference_t<Executor>>> &&
-            std::is_same_v<
-                decltype(std::declval<Executor>().execute(detail::noop)), void>,
-        decltype(std::declval<Executor&>() == std::declval<Executor&>(),
-                 (void)0)>> : std::true_type {};
+        std::is_copy_constructible<
+            execution::remove_cv_ref_t<Executor>>::value &&
+            std::is_same<
+                decltype(std::declval<execution::remove_cv_ref_t<Executor>>()
```

Comment:
I can understand the need to remove `const` if `Executor::execute` is not declared `const`, but what's the requirement behind removing references?

Reply:
Yup makes no sense to have it. Will reevaluate all of its uses.